Add async-std support and replaced reqwest with surf (#58)

JEnoch opened the pull request:
## Description

This PR replaces [reqwest](https://crates.io/crates/reqwest) usage with [surf](https://crates.io/crates/surf) usage that supports both `tokio` and `async-std`. It also modifies the tests to run with both.
Fixes #58. 

### Checklist
- [x] Formatted code using `cargo fmt --all`
- [x] Linted code using clippy `cargo clippy --all-targets --all-features -- -D warnings`
- [ ] Updated README.md using `cargo readme -r influxdb -t ../README.tpl > README.md`
- [x] Reviewed the diff. Did you leave any print statements or unnecessary comments?
- [x] Any unfinished work that warrants a separate issue captured in an issue with a TODO code comment

Empty2k12 commented:
Thanks for this PR!

Could we add two flags (`async-std` and `tokio`) and have tokio being the default one. If tokio is enabled, the hyper backend of surf is used, and if async-std is enabled, the curl backend is used? 

Also, I'd like to merge #73 first, so the changed calls can be ported over to surf as well.

JEnoch commented:
I'm not sure to understand the purpose of adding `async-std` and `tokio` features. Your influxdb-rust library is actually agnostic to `async-std` or `tokio`, as `surf` is as well.
I made some tests with `surf` and you mixing `hyper-client` or `curl-client` with either `async-std` or `tokio` for the binary `main()` works in all combinations. 

But you're right about giving the user the possibility to choose the http backend.
If you agree, I add in this PR the same features than `surf` in `influxdb-rust` (See https://github.com/http-rs/surf/blob/main/Cargo.toml#L19). And set `hyper-client` as default if you prefer.

JEnoch commented:
Hi @Empty2k12,

I updated this PR with latest status, merging with #73.
I also added the features to choose the http library used by `surf` (default to `hyper-client`), following my proposal above.

Note that switching to `surf` I get a 50% perf improvement on the bench pushed in #73 and no request errors (I had a lot when using reqwest) !

JEnoch commented:
Hi @Empty2k12 ,

Any concern or issue reviewing this PR ?

Empty2k12 commented:
Sorry, I'm quite busy with university right now. Tested your PR and will merge it now.
Thanks for this contribution! 🚀

JEnoch commented:
Thanks for the merge! (and sorry for the _"harassment"_ 😉 )
Do you plan a release and publication to crates.io any time soon ?

Empty2k12 commented:
@JEnoch Yes, I plan releasing this as 0.3.0.

Empty2k12 commented:
@JEnoch The version is now available under 0.3.0 at crates.io: https://crates.io/crates/influxdb

Thanks again for your contribution!

JEnoch commented:
@Empty2k12 That's great, thanks a lot! I just made eclipse-zenoh to use it (eclipse-zenoh/zenoh@a206fa83).
